crates/subspace-networking/src/protocols/connected_peers/tests.rs

```diff
@@ -0,0 +1,272 @@
+use crate::protocols::connected_peers::{Behaviour, Config, Event as ConnectedPeersEvent};
+use futures::{select, FutureExt};
+use libp2p::core::transport::MemoryTransport;
+use libp2p::core::upgrade::Version;
+use libp2p::core::Transport;
+use libp2p::identity::{Keypair, PeerId};
+use libp2p::plaintext::PlainText2Config;
+use libp2p::swarm::{NetworkBehaviour, SwarmBuilder, SwarmEvent};
+use libp2p::{yamux, Swarm};
+use libp2p_swarm_test::SwarmExt;
+use std::time::Duration;
+use tokio::time::sleep;
+
+#[derive(Debug, Clone)]
+struct ConnectedPeersInstance;
+
+#[tokio::test()]
+async fn test_connection_breaks_after_timeout_without_decision() {
+    let decision_timeout = Duration::from_millis(500);
+    let large_delay = Duration::from_millis(2000);
+
+    let mut peer1 = new_ephemeral(
+        decision_timeout,
+        Behaviour::<ConnectedPeersInstance>::new(Config {
+            decision_timeout,
+            ..Default::default()
+        }),
+    );
+
+    let mut peer2 = new_ephemeral(
+        decision_timeout,
+        Behaviour::<ConnectedPeersInstance>::new(Config {
+            decision_timeout,
+            ..Default::default()
+        }),
+    );
+
+    peer2.listen().await;
+    peer1.connect(&mut peer2).await;
+
+    loop {
+        select! {
+            _ = peer1.next_swarm_event().fuse() => {},
+            _ = peer2.next_swarm_event().fuse() => {},
+            _ = sleep(large_delay).fuse() => {
+                break;
+            }
+        }
+    }
+
+    assert!(!peer1.is_connected(peer2.local_peer_id()));
+    assert!(!peer2.is_connected(peer1.local_peer_id()));
+}
+
+#[tokio::test()]
+async fn test_connection_decision() {
```

Comment:
This and other tests will all run for 2 seconds, are you sure we can't make it run MUCH faster? Like 100ms decision timeout and large delay of 200ms?

Reply:
2 seconds are insignificant compared to the whole testing duration of the project. However, I can try setting it to 1s. Generally, I want to set delays to several orders of magnitude higher than the expected overhead of all the scaffolding.

Reply:
It is significant. Tests on my machine take ~5 minutes to run. 5 tests here will increase time to run tests by at least 2 seconds when running all in parallel, while not really using any meaningful amount of CPU. So having faster tests is certainly a desirable property. And in CI we only have 2 cores right now, so these 5 tests will take 6 seconds to run (depending on how they are scheduled). 3 seconds for 5 simple test cases is a lot.